app/service/learning_svc.py

```diff
@@ -5,6 +5,8 @@
 from importlib import import_module
 
 from app.objects.secondclass.c_relationship import Relationship
+from app.objects.secondclass.c_fact import OriginType
+from app.objects.secondclass.c_link import _update_scores
```

Comment:
Should we expose this properly if we are going to import it and use it elsewhere?

Reply:
also, should it be on the fact object if it really is central to there?

Reply:
I can get behind that. In theory, the learning parsers will be the only place this ever needs to be done, but it wouldn't hurt.

Reply:
I can't really get behind putting it in fact, though - it's largely part of a link's roles/duties, and just references and sets fact fields.

Reply:
ok, and that makes sense, I only saw it used twice, in c_link and here, so it is not widespread, it was just a linting red flag.